http-server/src/main/java/io/micronaut/servlet/http/server/HttpExchangeHttpServletRequest.java

```diff
@@ -0,0 +1,524 @@
+/*
+ * Copyright 2017-2025 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.servlet.http.server;
+
+import com.sun.net.httpserver.HttpExchange;
+import io.micronaut.core.annotation.Experimental;
+import io.micronaut.core.annotation.Internal;
+import io.micronaut.core.util.CollectionUtils;
+import io.micronaut.core.util.StringUtils;
+import io.micronaut.http.HttpHeaders;
+import io.micronaut.http.MediaType;
+import io.micronaut.http.cookie.ServerCookieDecoder;
+import io.micronaut.http.form.FormUrlEncodedDecoder;
+import io.micronaut.http.util.HttpHeadersUtil;
+import io.micronaut.servlet.http.utils.QueryStringDecoder;
+import jakarta.servlet.*;
+import jakarta.servlet.http.*;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.UnsupportedEncodingException;
+import java.nio.charset.StandardCharsets;
+import java.security.Principal;
+import java.util.*;
+
+/**
+ * Implementation of {@link HttpServletRequest} backed with a {@link HttpExchange}.
+ */
+@Experimental
+@Internal
+class HttpExchangeHttpServletRequest implements HttpServletRequest {
```

Comment:
wouldn't it be easier to implement Micronaut's HTTP request interface instead of trying to create a servlet bridge?